xmodule/block_metadata_utils.py

```diff
@@ -80,3 +86,22 @@ def display_name_with_default_escaped(block):
     # markupsafe.striptags() and fixing issues, better to put that energy toward
     # migrating away from this method altogether.
     return Markup(display_name_with_default(block)).striptags()
+
+
+def get_datetime_field(xblock, name, default):
```

Comment:
It looks like this is mostly copy/pasted from https://github.com/openedx/edx-platform/blob/3492bede446e4523ee09f965b7639b55af0472d0/openedx/core/djangoapps/content/block_structure/block_structure.py#L453-L483, is there a reason to redefine this instead of importing or using that function?

Reply:
looks like this is moving that part for reuse in block_structure and here, but maybe it could just use get_xblock_field directly as you say

Reply:
How do I re-use it? `self.due = subsection.transformer_data.get('due')`?

Reply:
neither `self` and `subsection` was a subclass of `BlockStructureBlockData`.

Reply:
After a discussion, decided that these are slightly different use-cases so it's okay to have a duplication, but should comment so it's clear that if one function changes both will probably have to change.